06/ukol10.py

```diff
@@ -0,0 +1,22 @@
+#Napiš funkci tah, která dostane řetězec s herním polem, číslo políčka (0-19), a symbol (x nebo o) a vrátí
+#herní pole (t.j. řetězec) s daným symbolem umístěným na danou pozici.
+#Hlavička funkce by tedy měla vypadat nějak takhle:
+#def tah(pole, cislo_policka, symbol):
+#"Vrátí herní pole s daným symbolem umístěným na danou pozici"
+
+def tah(field, position, symbol):
+    '''Vrati herni pole s danym symbolem umistenym na danou pozici.'''
+    start = field[:position]
+    end = field[position + 1:]
+    if (position >= 0 and position <= 19 and 
+        symbol == 'x' or symbol == 'o'):
+        change = start + symbol + end 
+        return change    
+    else:
+        return False    
```

Comment:
Je to správně - jen poznámečka, že pro kontrolu existence celého čísla se dá použít fajný mechanismus, který jsem rozepsal zde: https://github.com/Milis85/pyladies/pull/3#discussion_r1010212515

Jinak rozepsaný IF na víc řádků taky fajn - ale jak na to tak koukám, tak z hlediska čitelnosti (např. ani není přesně podle odsazení na první pohled rozpoznat, kde končí podmínka a kde začíná `True` blok toho IFu.

U složitějších podmínek nemusí být na škodu popřemýšlet nad tím, _"dát tu celou složenou podmínku do IFu, nebo si ty jednotlivé podmínky vyrefaktorovat nad to?"_

V tomhle případě by to mohlo vypadat třeba takto:

````python
def tah(field, position, symbol):
    '''Vrati herni pole s danym symbolem umistenym na danou pozici.'''
    correct_position = position in range(0, 20)
    correct_symbol = symbol == 'x' or symbol == 'o'
    if not correct_position or not correct_symbol):
        return False

    start = field[:position]
    end = field[position + 1:]
    return start + symbol + end 
```

- IF jsem tady zrefaktoroval tak, aby fungoval jako [early return](https://szymonkrajewski.pl/why-should-you-return-early/), což zpřehledňuje kód.
- A vyhodnocení hodnot ˙start˙ a `end` jsem strčil až na konec, protože pokud ani pozice a ani symbol není správný, tak se to vlastně nepoužilo - takže v tom případě bylo zbytečné to mít vyhodnocené předem.